classes/class-wpcom-liveblog-amp.php

```diff
@@ -188,50 +188,41 @@ public static function append_liveblog_to_metadata( $metadata, $post ) {
 
 		$request = self::get_request_data();
 
-		$publisher_organization = '';
-		$publisher_name         = '';
-
 		$entries = WPCOM_Liveblog::get_entries_paged( $request->page, $request->last );
 
 		$blog_updates = [];
 
-		if ( isset( $entries['entries'] ) && is_array( $entries['entries'] ) ) {
-			foreach ( $entries['entries'] as $key => $entry ) {
-
-				if ( isset( $metadata['publisher']['name'] ) ) {
-					$publisher_name = $metadata['publisher']['name'];
-				}
-
-				if ( isset( $metadata['publisher']['type'] ) ) {
-					$publisher_organization = $metadata['publisher']['type'];
-				}
-
-				$blog_item = (object) array(
-					'@type'         => 'BlogPosting',
-					'headline'      => self::get_entry_title( $entry ),
-					'url'           => $entry->share_link,
-					'datePublished' => date( 'c', $entry->entry_time ),
-					'dateModified'  => date( 'c', $entry->timestamp ),
-					'author'        => (object) array(
-						'@type' => 'Person',
-						'name'  => $entry->authors[0]['name'],
-					),
-					'articleBody'   => (object) array(
-						'@type' => 'Text',
-					),
-					'publisher'     => (object) array(
-						'@type' => $publisher_organization,
-						'name'  => $publisher_name,
-					),
-				);
-
-				array_push( $blog_updates, $blog_item );
+		if ( ! isset( $entries[ 'entries' ] ) || ! is_array( $entries[ 'entries' ] ) ) {
+			return $metadata;
+		}
+
+		foreach ( $entries[ 'entries' ] as $key => $entry ) {
+			$blog_item = [
+				'@type'            => 'BlogPosting',
+				'headline'         => self::get_entry_title( $entry ),
+				'url'              => $entry->share_link,
+				'mainEntityOfPage' => $entry->share_link,
+				'datePublished'    => date( 'c', $entry->entry_time ),
+				'dateModified'     => date( 'c', $entry->timestamp ),
+				'author'           => [
+					'@type' => 'Person',
+					'name'  => $entry->authors[ 0 ][ 'name' ],
+				],
+				'articleBody'      => [
+					'@type' => 'Text',
+				],
+			];
+
+			if ( isset( $metadata['publisher'] ) ) {
+				$blog_item['publisher'] = $metadata['publisher'];
 			}
 
-			$metadata['@type']          = 'LiveBlogPosting';
-			$metadata['liveBlogUpdate'] = $blog_updates;
+			$blog_updates[] = json_decode( json_encode( $blog_item ) );
```

Comment:
Can you elaborate on the need for this change please? I'm not sure I understand what it adds.

Reply:
The previous code was casting each array to an `(object)`, which makes adding and amending data in the array (such as a conditional publisher logo) awkward. This conversion should be a one-time recursive task. `json_decode( json_encode() )` is a short-hand way of doing just that. This could be pulled into a separate method (such as `convertToObjects()`, or a better domain-related name) once the abstraction was done.